youtube_dl/downloader/ism.py

```diff
@@ -228,7 +228,7 @@ def real_download(self, filename, info_dict):
         frag_index = 0
         for i, segment in enumerate(segments):
             frag_index += 1
-            if frag_index <= ctx['fragment_index']:
+            if frag_index <= ctx['fragment_index'] or self._should_skip_fragment(info_dict, segment['url']):
```

Comment:
Split into two `if` statements for uniformity with other places in code and better readability.